lib/iris/fileformats/grib/__init__.py

```diff
@@ -940,7 +940,57 @@ def save_grib2(cube, target, append=False, **kwargs):
     See also :func:`iris.io.save`.
 
     """
+    messages = as_messages(cube)
+    save(messages, target, append=append)
 
+
+def as_pairs(cube):
+    """
+    Convert one or more cubes to (2D cube, GRIB message) pairs.
+
+    Args:
+        * cube      - A :class:`iris.cube.Cube`, :class:`iris.cube.CubeList` or list of cubes.
+
+    """
+    x_coords = cube.coords(axis='x', dim_coords=True)
+    y_coords = cube.coords(axis='y', dim_coords=True)
+    if len(x_coords) != 1 or len(y_coords) != 1:
+        raise TranslationError("Did not find one (and only one) x or y coord")
+
+    # Save each latlon slice2D in the cube
+    for slice2D in cube.slices([y_coords[0], x_coords[0]]):
+        grib_message = gribapi.grib_new_from_samples("GRIB2")
+        _save_rules.run(slice2D, grib_message)
+        yield (slice2D, grib_message)
+
+
+def as_messages(cube):
+    """
+    Convert one or more cubes to GRIB messages.
+
+    Args:
+        * cube      - A :class:`iris.cube.Cube`, :class:`iris.cube.CubeList` or list of cubes.
+
+    """
+    return (message for cube, message in as_pairs(cube))
+
+
+def save(messages, target, append=False):
```

Comment:
Suggestion: rename to `save_messages`.

Rationale: We need a pattern of naming for GRIB and PP. Since PP uses `as_fields` and `save_fields`, then I suggest GRIB should use `as_messages` and `save_messages`.